src/tasks/AppleAppBuilder/Templates/runtime-coreclr.m

```diff
@@ -0,0 +1,129 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#import <Foundation/Foundation.h>
+#include <coreclrhost.h>
+#include <TargetConditionals.h>
+#import <os/log.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+static char *bundle_path;
+
+#define APPLE_RUNTIME_IDENTIFIER "//%APPLE_RUNTIME_IDENTIFIER%"
+
+const char *
+get_bundle_path (void)
+{
+    if (bundle_path)
+        return bundle_path;
+    NSBundle* main_bundle = [NSBundle mainBundle];
+    NSString* path = [main_bundle bundlePath];
+
+#if TARGET_OS_MACCATALYST
+    path = [path stringByAppendingString:@"/Contents/Resources"];
+#endif
+
+    bundle_path = strdup ([path UTF8String]);
+
+    return bundle_path;
+}
+
+char *
+strdup_printf (const char *msg, ...)
+{
+    va_list args;
+    char *formatted = NULL;
+    va_start (args, msg);
+    vasprintf (&formatted, msg, args);
+    va_end (args);
+    return formatted;
+}
+
+void
+log_callback (const char *log_domain, const char *log_level, const char *message, bool fatal, void *user_data)
+{
+    os_log_info (OS_LOG_DEFAULT, "(%s %s) %s", log_domain, log_level, message);
+    if (fatal) {
+        os_log_info (OS_LOG_DEFAULT, "Exit code: %d.", 1);
+        exit (1);
+    }
+}
+
+void
+mono_ios_runtime_init (void)
+{
+#if INVARIANT_GLOBALIZATION
+    setenv ("DOTNET_SYSTEM_GLOBALIZATION_INVARIANT", "1", TRUE);
+#endif
+
+    id args_array = [[NSProcessInfo processInfo] arguments];
+    assert ([args_array count] <= 128);
+    const char *managed_argv [128];
+    int argi;
+    for (argi = 0; argi < [args_array count]; argi++) {
+        NSString* arg = [args_array objectAtIndex: argi];
+        managed_argv[argi] = [arg UTF8String];
+    }
+
+    const char* bundle = get_bundle_path ();
+    chdir (bundle);
```

Comment:
Nice hack. 🤣 

it'll break probably as soon as any test changes the working directory.  but it's a good place to start.

Reply:
oh I see you added the TPA list.  nevermind.

Reply:
Talk to Egor, that's lifted wholesale from the existing Mono runtime.m

https://github.com/dotnet/runtime/blame/main/src/tasks/AppleAppBuilder/Templates/runtime.m#L236-L237